source/extensions/filters/http/lua/lua_filter.h

```diff
@@ -309,8 +311,8 @@ class Filter : public Http::StreamFilter, Logger::Loggable<Logger::Id::lua> {
     return Http::FilterHeadersStatus::Continue;
   }
   Http::FilterHeadersStatus encodeHeaders(Http::HeaderMap& headers, bool end_stream) override {
-    return doHeaders(response_stream_wrapper_, encoder_callbacks_, config_->responseFunctionRef(),
-                     headers, end_stream);
+    return doHeaders(response_stream_wrapper_, response_coroutine_, encoder_callbacks_,
```

Comment:
not related to this PR, but  just for my knowledge, would it makes sense here to just pass a `bool` indicating whether it is `encode` or `decode` instead and let `doHeaders` pickup the right ones (since they are available at class level), instead of passing all of them around?

Reply:
I think it would be fine either way, but I'm going to leave it as is.